Sources/Purchasing/OfferingsManager.swift

```diff
@@ -24,42 +24,77 @@ class OfferingsManager {
     private let backend: Backend
     private let offeringsFactory: OfferingsFactory
     private let productsManager: ProductsManagerType
+    private let diagnosticsTracker: DiagnosticsTrackerType?
+    private let dateProvider: DateProvider
 
     init(deviceCache: DeviceCache,
          operationDispatcher: OperationDispatcher,
          systemInfo: SystemInfo,
          backend: Backend,
          offeringsFactory: OfferingsFactory,
-         productsManager: ProductsManagerType) {
+         productsManager: ProductsManagerType,
+         diagnosticsTracker: DiagnosticsTrackerType?,
+         dateProvider: DateProvider = DateProvider()) {
         self.deviceCache = deviceCache
         self.operationDispatcher = operationDispatcher
         self.systemInfo = systemInfo
         self.backend = backend
         self.offeringsFactory = offeringsFactory
         self.productsManager = productsManager
+        self.diagnosticsTracker = diagnosticsTracker
+        self.dateProvider = dateProvider
     }
 
     func offerings(
         appUserID: String,
         fetchPolicy: FetchPolicy = .default,
         fetchCurrent: Bool = false,
+        trackDiagnostics: Bool = true,
```

Comment:
I had doubts about whether to add this... This method is also used upon storefront changes, so not when the developer calls any of the public APIs to get offerings. I didn't want to move the logic outside of this method though, since that would be much trickier and have more repetition, so I ended up adding a boolean parameter to disable tracking in that case.